xmake/modules/package/tools/meson.lua

```diff
@@ -165,113 +269,15 @@ function _get_cross_file(package, opt)
             file:print("c_link_args=['%s']", table.concat(linkflags, "', '"))
             file:print("cpp_link_args=['%s']", table.concat(linkflags, "', '"))
```

Comment:
we should set ant cc/ld for native host.

if you want to support `--toolchain`, you can refer 

https://github.com/xmake-io/xmake/blob/04f15ffea7ba45d06be9d0a69bad1e1e8509ccd0/xmake/modules/package/tools/cmake.lua#L719-L729

Reply:
What do you mean by "ant" ?

Reply:
ant?

Reply:
typo. sorry. we should not set cc/ld for native host.

Reply:
Why not ? Even with cmake the path to the host toolchain is given.
https://github.com/xmake-io/xmake/blob/8fcc525666412aefa0457bf866922b8e97818ea8/xmake/modules/package/tools/cmake.lua#L573-L579
And if the path to the correct tooolchain is not set meson keeps using the default detected one. 
Example :
On Windows a project set with a llvm toolchain, meson will use msvc if we didn't tell to do so. I have not tested on linux, but I think it's the same problem, if the sdk was set the another path that the default system toolchain meson will use the wrong one.

Reply:
> Why not ? Even with cmake the path to the host toolchain is given.

Only when toolchain is specified explicitly, because it changes the default host compiler.

```console
xmake f --toolchain=clang
```

or `set_toolchains("clang")`

But if toolchains are not explicitly set, we only need to use the default system compiler and do not need to explicitly specify cc. Neither cmake nor autoconf specifies it.


https://github.com/xmake-io/xmake/blob/8fcc525666412aefa0457bf866922b8e97818ea8/xmake/modules/package/tools/cmake.lua#L721

 > On Windows a project set with a llvm toolchain, meson will use msvc if we didn't tell to do so. I have not tested on linux, but I think it's the same problem, if the sdk was set the another path that the default system toolchain meson will use the wrong one.

so you changed toolchain, you can set cc/ld.

```lua
    elseif package:config("toolchains") then
        -- we still need find system libraries,
        -- it just pass toolchain environments if the toolchain is compatible with host
        if _is_toolchain_compatible_with_host(package) then
            _get_configs_for_host_toolchain(package, configs, opt)
        else
            _get_configs_for_cross(package, configs, opt)
        end
    else
```

But your patch will always modify cc/ld , even if we don't set toolchains explicitly.

Reply:
We should judge `package:config("toolchains")` and only modify cc/ld when it is not empty